server/controller/provider/getProviderData.js

```diff
@@ -0,0 +1,17 @@
+const { getProviderDataById } = require('../../database/queries/provider');
+
+const providerData = async (req, res, next) => {
+  try {
+    const { id } = req.params;
```

Comment:
no need to get the id value from param since you have the provider id on middleware ``` req.user.id ```